scripts/Muon/GUI/Common/fitting_tab_widget/fitting_tab_presenter.py

```diff
@@ -508,13 +429,132 @@ def clear_fit_information(self):
         self.update_fit_status_information_in_view()
         self.view.undo_fit_button.setEnabled(False)
 
-    @property
-    def start_x(self):
-        return self._start_x
+    def update_selected_workspace_guess(self):
+        if self.view.is_simul_fit():
+            self.update_fit_specifier_list()
+        if self.manual_selection_made:
+            guess_selection = self.selected_data
+            self.selected_data = guess_selection
+        else:
+            self.selected_data = self.get_workspace_selected_list()
 
-    @property
-    def end_x(self):
-        return self._end_x
+    def get_workspace_selected_list(self):
+        if self.context._frequency_context is not None:
+            freq = self.context._frequency_context.plot_type
+        else:
+            freq = 'None'
+
+        runs = 'All'
+        groups_and_pairs = self._get_selected_groups_and_pairs()
+        if self.view.is_simul_fit():
+            if self.view.simultaneous_fit_by == "Run":
+                runs = self.view.simultaneous_fit_by_specifier
+            elif self.view.simultaneous_fit_by == "Group/Pair":
+                groups_and_pairs = [self.view.simultaneous_fit_by_specifier]
+
+        guess_selection = []
+        for grppair in groups_and_pairs:
```

Comment:
I dont know what `grppair` is supposed to be

Reply:
The intention was this line is read as, "for group_or_pair in groups_and_pairs". I'll be more clearer with the naming though and change this.